app.js

```diff
@@ -11,19 +11,31 @@ var logger = require('morgan');
 var bodyParser = require('body-parser');
 var mongoose = require('mongoose');
 var router = require('./routes/router');
+var scr = require('./controllers/repositories/systemConfigRepository');
 
 
 var contextRoot = "/";  //Not set any contextRoot at the moment, but let's make it as easy to config
 
 
-var	mongoURL = 'mongodb://127.0.0.1:27017/camps';
-
-mongoose.connect(mongoURL, function(err) {
-	if (err) {
-		console.log("could not connect mongodb with error message "+ err)
-	}
+scr.getMongoCredentials().then(function(data){
```

Comment:
"then" here, is only indicating successful callback. How about fail? say the secret file is not put correctly.